modules/sievefilters/modules.php

```diff
@@ -1379,11 +1379,67 @@ public function process() {
             save_main_script($client, $main_script, $scripts);
             $client->activateScript('main_script');
             $client->close();
-            
+
             Hm_Msgs::add("Script $state");
         } catch (Exception $e) {
             Hm_Msgs::add("ERRSieve: {$e->getMessage()}");
         }
         $this->out('success', $success);
     }
 }
+class Hm_Handler_list_block_sieve_script extends Hm_Handler_Module {
+    public function process() {
+        list($success, $form) = $this->process_form(array('imap_server_id'));
+        if (!$success) {
+            return;
+        }
+
+        foreach ($this->user_config->get('imap_servers') as $idx => $mailbox) {
```

Comment:
Accessing internal representation of user config in unrelated modules is not recommended. Better use Hm_Imap_List::get() method to get the details by server ID.

Reply:
This doesn't seem resolved.